hack/cloud-image-uploader/azure-image-version-template.json

```diff
@@ -0,0 +1,61 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+      "gallery_name": {
+        "defaultValue": null,
+        "type": "String"
+      },
+      "disk_name": {
+        "defaultValue": null,
+        "type": "String"
+      },
+      "image_version": {
+        "defaultValue": null,
+        "type": "String"
+      },
+      "definition_name": {
+        "defaultValue": null,
+        "type": "String"
+      },
+      "region": {
+        "defaultValue": null,
+        "type": "String"
+      },
+      "resourceGroupName": {
+        "type": "String"
+      },
+      "targetRegions": {
+        "type": "array"
+      }
+    },
+    "variables": {},
+    "resources": [
+      {
+        "type": "Microsoft.Compute/galleries/images/versions",
+        "apiVersion": "2022-03-03",
+        "name": "[concat(parameters('gallery_name'), '/', parameters('definition_name'), '/', parameters('image_version'))]",
+        "location": "[parameters('region')]",
+        "properties": {
+          "publishingProfile": {
+            "targetRegions": "[parameters('targetRegions')]",
+            "replicaCount": 1,
+            "excludeFromLatest": false,
+            "storageAccountType": "Standard_LRS",
+            "replicationMode": "Full"
+          },
+          "storageProfile": {
+            "osDiskImage": {
+              "hostCaching": "None",
+              "source": {
+                "id": "[concat('/subscriptions/7f739b7d-f399-4b97-9a9f-f1962309ee6e/resourceGroups/SIDEROGALLERY/providers/Microsoft.Compute/disks/', parameters('disk_name'))]"
```

Comment:
this mix of hardcoded and provided variables feels like a weird mix to me.

I think we could have these files, `//go:embed` them, and use Go templating to populate all values, this will be more readable/maintainable in the future.

Reply:
The way these arm templates are written are exactly the way Azure wants them formatted. When deploymentsClient.BeginCreateOrUpdate is called, it uses the variables passed in with it to fill in the parameters. I can look into using go:embed, but maintaining these, only requires you to log on to the Portal and grab a new example template from an existing resource with updated API versions or syntax.